analyses/snv-callers/scripts/03-merge_callers.R

```diff
@@ -0,0 +1,305 @@
+all_vaf_file <- file.path(opt$output, "all_callers_vaf.rds")
+all_tmb_file <- file.path(opt$output, "all_callers_tmb.rds")
+mut_id_file <- file.path(opt$output, "mutation_id_list.rds")
+call_per_mut_file <- file.path(opt$output, "callers_per_mutation.rds")
+
+##################### Check for files if overwrite is FALSE ####################
+# If overwrite is set to FALSE, check if these exist before continuing
+if (!opt$overwrite) {
+  # Make a list of the output files
+  output_files <- c(all_vaf_file, all_tmb_file, mut_id_file, call_per_mut_file)
+
+  # Find out which of these exist
+  existing_files <- file.exists(output_files)
+
+  # If all files exist; stop
+  if (all(existing_files)) {
+    stop(cat(
+      "Stopping; --overwrite is not being used and all output files already exist
+      in the designated --output directory."
+    ))
+  }
+  # If some files exist, print a warning:
+  if (any(existing_files)) {
+    warning(cat(
+      "Some output files already exist and will not be overwritten unless you use --overwrite: \n",
+      paste0(output_files[which(existing_files)], "\n")
+    ))
+  }
+}
+
+########################### Make Master VAF file ###############################
+# If the file exists or the overwrite option is not being used, do not write the
+# merged VAF file.
+if (file.exists(all_vaf_file) && !opt$overwrite) {
+  # Stop if this file exists and overwrite is set to FALSE
+  warning(cat(
+    "The merged VAF file already exists: \n",
+    all_vaf_file, "\n",
+    "Use --overwrite if you want to overwrite it."
+  ))
+} else {
+  # Get the caller names
+  caller_names <- stringr::word(vaf_files, sep = "/", -2)
+
+  # Read in vaf files for all callers
+  if (opt$file_format == "tsv") {
+    vaf_list <- lapply(vaf_files, readr::read_tsv)
+  } else {
+    vaf_list <- lapply(vaf_files, readr::read_rds)
+  }
+
+  # Read in the other files to match the first
+  vaf_list <- lapply(vaf_list, function(df) {
+    # Make it so it is more easily combined with the other files
+    df %>%
+      # Attempt to make numeric columns where that doesn' kick back an "NA"
+      dplyr::mutate_at(dplyr::vars(which(!is.na(as.numeric(t(df[1, ]))))), as.numeric) %>%
+      # Aliquot id sometimes contains letters and sometimes numbers across the callers
+      dplyr::mutate(
+        aliquot_id = as.character(aliquot_id),
+        variant_qual = as.character(variant_qual)
+      ) %>%
+      # Turn these columns into characters because otherwise they cause trouble.
+      dplyr::mutate_at(dplyr::vars(dplyr::contains("AF", ignore.case = FALSE)), as.character) %>%
+      # Get rid of the few if any duplicate entries.
+      dplyr::distinct(mutation_id, .keep_all = TRUE)
+  })
+
+  # Carry over the callers' names
+  names(vaf_list) <- caller_names
+
+  # Print progress message
+  message("Saving master VAF file to: \n", all_vaf_file)
+
+  # Combine and save VAF file
+  vaf_df <- suppressWarnings(dplyr::bind_rows(vaf_list, .id = "caller")) %>%
```

Comment:
Can you put a detailed comment here explaining why you're using `suppressWarnings` and why the warnings are expected?